sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -858,3 +860,60 @@ def merge_only(self, accumulators):
 
   def extract_only(self, accumulator):
     return self.combine_fn.extract_output(accumulator)
+
+
+class Latest(object):
+  """Combiners for computing the latest element"""
+
+  class Globally(ptransform.PTransform):
+    """Compute the element with the latest timestamp from a
+    PCollection."""
+
+    @staticmethod
+    def add_timestamp(element, timestamp=core.DoFn.TimestampParam):
+      return [(element, timestamp)]
+
+    def expand(self, pcoll):
+      return (pcoll
+              | core.ParDo(self.add_timestamp)
+              | core.CombineGlobally(LatestCombineFn()))
+
+  class PerKey(ptransform.PTransform):
+    """Compute elements with the latest timestamp for each key
+    from a keyed PCollection"""
+
+    @staticmethod
+    def add_timestamp(element, timestamp=core.DoFn.TimestampParam):
+      _check_instance_type(KV[T, T], element)
+      (K, V) = element
+      return [(K, (V, timestamp))]
+
+    def expand(self, pcoll):
+      return (pcoll
+              | core.ParDo(self.add_timestamp)
+              | core.CombinePerKey(LatestCombineFn()))
+
+
+@with_input_types(KV[T, T])
```

Comment:
Using KV as type hint seems a bit weird to me because the input is really a (element, timestamp) tuple. Maybe we can just use something like Tuple[E, T] as input type and E as output type. For more about Python type hints, see: https://beam.apache.org/documentation/sdks/python-type-safety/

Reply:
I agree. It should be a Tuple rather than a KV.

Also, only types int, float, long, Timestamp and Duration can be interpreted as timestamps[1] so they can be compared to `window.MIN_TIMESTAMP` in the accumulator. So, having the timestamp's type as T would be less restrictive. 
Does it make sense to create a new type - `TimestampT/TimestampType = Union[int, float, long, Timestamp, Duration]` in typehints.py or the combiners module? This would make the input type `Tuple[T, TimestampType]`.

[1] https://github.com/apache/beam/blob/master/sdks/python/apache_beam/utils/timestamp.py#L71

Reply:
That's a good suggestion. We can add the definition in this module.
Why is Duration also allowed?

Reply:
Both `Timestamp` and `Duration` have an attribute - `micros`. This is used to compare the two.

https://github.com/apache/beam/blob/master/sdks/python/apache_beam/utils/timestamp.py#L161-L165